connectors/src/connectors/slack/temporal/activities.ts

```diff
@@ -0,0 +1,339 @@
+  slackAccessToken: string
+): Promise<Channel[]> {
+  const client = new WebClient(slackAccessToken);
+  const allChannels = [];
+  let nextCursor: string | undefined = undefined;
+  do {
+    const c: ConversationsListResponse = await client.conversations.list({
+      types: "public_channel,private_channel",
+      limit: 1000,
+      cursor: nextCursor,
+    });
+    nextCursor = c?.response_metadata?.next_cursor;
+    if (c.error) {
+      throw new Error(c.error);
+    }
+    if (!c.channels) {
+      throw new Error(
+        "There was no channels in the response for cursor " +
+          c?.response_metadata?.next_cursor +
+          ""
+      );
+    }
+    for (const channel of c.channels) {
+      if (channel && channel.id && channel.is_member) {
+        allChannels.push(channel);
+      }
+    }
+  } while (nextCursor);
+
+  return allChannels;
+}
+
+export async function getMessagesForChannel(
+  slackAccessToken: string,
+  channelId: string,
+  limit = 100,
+  nextCursor?: string
+): Promise<ConversationsHistoryResponse> {
+  const client = new WebClient(slackAccessToken);
+
+  const c: ConversationsHistoryResponse = await client.conversations.history({
+    channel: channelId,
+    limit: limit,
+    cursor: nextCursor,
+  });
+  if (c.error) {
+    throw new Error(
+      `Failed getting messages for channel ${channelId}: ${c.error}`
+    );
+  }
+
+  return c;
+}
+
+export async function syncMultipleNoNThreaded(
+  slackAccessToken: string,
+  dataSourceConfig: DataSourceConfig,
+  channelId: string,
+  channelName: string,
+  timestampsMs: { startTsMs: number; endTsMs: number }[]
+) {
+  while (timestampsMs.length > 0) {
+    const _timetampsMs = timestampsMs.splice(0, MAX_CONCURRENCY_LEVEL);
+
+    await Promise.all(
+      _timetampsMs.map((t) =>
+        syncNonThreaded(
+          slackAccessToken,
+          dataSourceConfig,
+          channelId,
+          channelName,
+          t.startTsMs,
+          t.endTsMs
+        )
+      )
+    );
+  }
+}
+
+export async function syncNonThreaded(
+  slackAccessToken: string,
+  dataSourceConfig: DataSourceConfig,
+  channelId: string,
+  channelName: string,
+  startTsMs: number,
+  endTsMs: number
+) {
+  const client = new WebClient(slackAccessToken);
+  const nextCursor: string | undefined = undefined;
+  const messages: Message[] = [];
+
+  const startTsSec = Math.round(startTsMs / 1000);
+  const endTsSec = Math.round(endTsMs / 1000);
+
+  const c: ConversationsHistoryResponse = await client.conversations.history({
+    channel: channelId,
+    limit: 300,
+    oldest: `${startTsSec}`,
+    latest: `${endTsSec}`,
+    cursor: nextCursor,
+  });
+  if (c.error) {
+    throw new Error(
+      `Failed getting messages for channel ${channelId}: ${c.error}`
+    );
+  }
+  if (c.messages === undefined) {
+    throw new Error(
+      `Failed getting messages for channel ${channelId}: messages is undefined`
+    );
+  }
+  for (const message of c.messages) {
```

Comment:
That does not include threaded messages? Even initial ones? Even ones posted to channel?